File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int i =0, j = s.size() -1;
        string vowels = "aeiouAEIOU";
        while ( i< j) {
         if (vowels.find(s[i]) == string :: npos) i++;

         else if (vowels.find(s[j]) == string :: npos) j--;

         else {
            swap(s[i], s[j]);
            i++;
            j--;
         }
        }
        return s;
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        arr = list(s)
        i = 0
        j = len(arr) -1
        while i < j:
            if arr[i].lower() in "aeiou":
                if arr[j].lower() in "aeiou":
                    arr[i], arr[j] = arr[j], arr[i]
                    i += 1
                    j -= 1
                else:
                    j -= 1
            else:
                i += 1    
        return "".join(arr)  
        